packages/hint-doctype/src/hint.ts

```diff
@@ -30,13 +30,12 @@ export default class implements IHint {
     }
 
     public constructor(context: HintContext) {
-        const regExpFactory = (isFlexible = false) => {
-            return new RegExp(`(<!doctype\\s+(html)\\s*(\\s+system\\s+"about:legacy-compat")?\\s*?>)${isFlexible ? '(.+)?' : ''}`, 'gi');
+        const regExpFactory = () => {
```

Comment:
Is this still necessary? You are using it just once, so maybe it is better to have something like `const doctypeRegExp = /myregex/gi`